gift-certificate-application/controller/src/main/java/com/epam/esm/controller/exception/AppExceptionHandler.java

```diff
@@ -0,0 +1,150 @@
+package com.epam.esm.controller.exception;
+
+import com.epam.esm.service.exception.*;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import static java.time.LocalDateTime.now;
+
+@RestControllerAdvice
+public class AppExceptionHandler {
+    private static final int FIELD_VALIDATION_ERROR_CODE = 40001;
+    private static final int GIFT_CERTIFICATE_ALREADY_EXIST_ERROR_CODE = 40002;
+    private static final int TAG_ALREADY_EXIST_ERROR_CODE = 40003;
+    private static final int TAG_TO_GIFT_CERTIFICATE_REFERENCE_ERROR_CODE = 40004;
+    private static final int INVALID_COLUMN_NAME_ERROR_CODE = 40005;
+    private static final int INVALID_SORT_TYPE_ERROR_CODE = 40006;
+    private static final int USER_ALREADY_EXIST_ERROR_CODE = 40007;
+    private static final int HAS_ORDER_TO_GIFT_CERTIFICATE_ERROR_CODE = 40008;
+    private static final int TAG_NOT_FOUND_ERROR_CODE = 40401;
+    private static final int GIFT_CERTIFICATE_NOT_FOUND_ERROR_CODE = 40402;
+    private static final int GIFT_CERTIFICATES_NOT_FOUND_ERROR_CODE = 40403;
+    private static final int CANNOT_UPDATE_ERROR_CODE = 40404;
+    private static final int USER_HAS_NO_ORDER_ERROR_CODE = 40405;
+    private static final int USER_NOT_FOUND_ERROR_CODE = 40406;
+    private static final int ORDER_NOT_FOUND_ERROR_CODE = 40407;
+    private static final int TAG_DOES_NOT_EXIST_ERROR_CODE = 40408;
+    private static final int NO_GIFT_CERTIFICATE_MATCHING_ERROR_CODE = 40409;
+
+    public ErrorResponse handleCommonException(Exception exception, HttpStatus status, int errorCode) {
```

Comment:
I suppose that this method is only used in this class, so make it private in move at the end of the class